core/utils.js

```diff
@@ -99,33 +99,39 @@ Blockly.hasClass_ = function(element, className) {
  * @private
  */
 Blockly.bindEvent_ = function(node, name, thisObject, func) {
-  if (thisObject) {
-    var wrapFunc = function(e) {
-      func.call(thisObject, e);
-    };
-  } else {
-    var wrapFunc = func;
-  }
+  var wrapFunc = function(e) {
+    // Handle each touch point separately.  If the event was a mouse event, this
+    // will hand back an array with one element, which we're fine handling.
+    var events = Blockly.bindEvent_.splitEventByTouches(e);
+    for (var i = 0; i < events.length; i++) {
+      var event = events[i];
```

Comment:
Simpler (assuming there isn't a falsy value in the events list):
for (var i = 0, event; event = events[i]; i++) {